packages/auth/src/Expo-AuthSession/useExpoAuthenticate.ts

```diff
@@ -1,85 +1,131 @@
+        if (!authState.discovery) authState.setDiscovery(discovery);
+    }, [discovery]);
+
+    const authenticateInteractively = async () => {
         try {
             setAuthenticationInProgress(true);
-            const codeResponse = await promptAsync();
-            if (request && codeResponse?.type === "success" && discovery) {
-                const result = await exchangeCodeAsync(
-                    {
-                        clientId,
-                        code: codeResponse.params["code"],
-                        extraParams: request.codeVerifier
-                            ? { code_verifier: request.codeVerifier }
-                            : undefined,
-                        redirectUri,
-                    },
-                    discovery,
-                );
-
-                const user = decodeToken(result.accessToken);
-                if (!user) throw new Error("Unable to decode id token");
-                setAuthenticationInProgress(false);
-                onAuthenticationSuccessful(
-                    {
-                        accessToken: result.accessToken,
-                        account: user,
-                    },
-                    "MANUAL",
-                );
-            }
+            const result = await authenticate();
+            const user = decodeToken(result?.accessToken);
+            if (!result || !user) throw new Error("Unable to authenticate");
+            authState.setUserData(user);
+            authState.setToken(result);
+            setAuthenticationInProgress(false);
+            setLoginSuccess(result, user);
         } catch (error) {
             console.error(error);
             setAuthenticationInProgress(false);
             onAuthenticationFailed(error);
         }
     };
 
+    const authenticate = async () => {
+        const codeResponse = await promptAsync();
+        if (
+            request &&
+            codeResponse?.type === "success" &&
+            authState.discovery &&
+            authState.config
+        ) {
+            return await exchangeCodeAsync(
+                {
+                    clientId: authState.config?.clientId,
+                    code: codeResponse.params["code"],
+                    extraParams: request.codeVerifier
+                        ? { code_verifier: request.codeVerifier }
+                        : undefined,
+                    redirectUri: authState.config?.redirectUri,
+                },
+                authState.discovery,
+            );
+        }
+        throw new Error("Unable to authenticate");
+    };
+
+    const refreshTokenAndAuthenticate = async (token: TokenResponse, userData: MadAccount) => {
+        if (!authState.discovery || !authState.config) return;
+        const newToken = await refreshToken(token, userData);
+        if (newToken) {
+            authState.setToken(newToken);
+            setLoginSuccess(newToken, userData);
```

Comment:
Do we have access to `authenticateSilently` without using a hook?

Reply:
the new authenticateSilently is `getToken` and from `authStore.ts` I noticed now that `authenticateSilently` also returns the account. I think I would prefer adding a `getUserData` for returning the account and having `getToken` and `getUserData` replace the old `authenticateSilently` method. Do you agree?